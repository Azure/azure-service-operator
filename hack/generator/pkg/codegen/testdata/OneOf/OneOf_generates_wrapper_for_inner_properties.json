{
    "$comment": "Here we check that optional properties should generate pointers in the result",
    "id": "https://test.test/schemas/2020-01-01/test.json",
    "$schema": "http://json-schema.org/draft-04/schema#",
    "title": "Test",
    "type": "object",
    "properties": {
        "myString": {
            "type": "string"
        },
        "myBool": {
            "type": "boolean"
        },
        "properties": {
            "oneOf": [
                {
                    "$ref": "#/definitions/Foo"
                },
                {
                    "$ref": "#/definitions/Bar"
                }
            ]
        }
    },
    "definitions": {
        "Foo": {
            "type": "object",
            "properties": {
                "name": {
                    "type": "string"
                }
            }
        },
        "Bar": {
            "type": "object",
            "properties": {
                "size": {
                    "type": "integer"
                }
            },
            "required": ["size"]
        }
    }
}