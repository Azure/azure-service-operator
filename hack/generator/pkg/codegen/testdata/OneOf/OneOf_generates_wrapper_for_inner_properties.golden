// Code generated by k8s-infra. DO NOT EDIT.
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.
package v1alpha1api20200101

//Generated from: https://test.test/schemas/2020-01-01/test.json
type Test struct {
	MyBool     *bool            `json:"myBool,omitempty"`
	MyString   *string          `json:"myString,omitempty"`
	Properties *Test_Properties `json:"properties,omitempty"`
}

type Test_Properties struct {
	//Bar: Mutually exclusive with all other properties
	Bar *Bar `json:"bar,omitempty"`

	//Foo: Mutually exclusive with all other properties
	Foo *Foo `json:"foo,omitempty"`
}

//Generated from: https://test.test/schemas/2020-01-01/test.json#/definitions/Bar
type Bar struct {
	// +kubebuilder:validation:Required
	Size int `json:"size"`
}

//Generated from: https://test.test/schemas/2020-01-01/test.json#/definitions/Foo
type Foo struct {
	Name *string `json:"name,omitempty"`
}
